daemons/controld/controld_membership.c

```diff
@@ -287,6 +287,61 @@ search_conflicting_node_callback(xmlNode * msg, int call_id, int rc,
     }
 }
 
+/*!
+ * \internal
+ * \brief Populate a \c PCMK_XE_NODES element from the peer cache
+ *
+ * Create a \c PCMK_XE_NODE element for each node in the cache.
+ *
+ * \param[in,out] nodes_xml  \c PCMK_XE_NODES element to populate
+ */
+static void
+populate_cib_nodes_from_cache(xmlNode *nodes_xml)
+{
+    GString *xpath = NULL;
+    GHashTableIter iter;
+    pcmk__node_status_t *node = NULL;
+
+    g_hash_table_iter_init(&iter, pcmk__peer_cache);
+    while (g_hash_table_iter_next(&iter, NULL, (gpointer *) &node)) {
+        cib_t *cib_conn = controld_globals.cib_conn;
```

Comment:
This could be lifted out of the while loop.

Reply:
So could `call_id` and `new_node`. Yeah it gets assigned the same value every time, but there's no overhead to speak of, and the idea was to limit it to the narrowest scope possible. I prefer to do that in general.

Incidentally these could all go inside the `if`, but we de-nest in the next commit.